src/cpu/aarch64/acl_inner_product.cpp

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+* Copyright 2021 Arm Ltd. and affiliates
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "cpu/aarch64/acl_inner_product.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace aarch64 {
+
+using namespace dnnl::impl::status;
+using namespace dnnl::impl::memory_tracking::names;
+using namespace dnnl::impl::utils;
+
+status_t acl_inner_product_fwd_t::execute_forward(const exec_ctx_t &ctx) const {
+    status_t status = status::success;
+    auto src_base = CTX_IN_MEM(const data_t *, DNNL_ARG_SRC);
+    auto wei_base = CTX_IN_MEM(const data_t *, DNNL_ARG_WEIGHTS);
+    auto bia_base = CTX_IN_MEM(const data_t *, DNNL_ARG_BIAS);
+    auto dst_base = CTX_OUT_MEM(data_t *, DNNL_ARG_DST);
+
+    bool with_bias = pd()->aip_.with_bias;
+    bool with_sum = pd()->aip_.with_sum;
+
+    // Retrieve primitive resource and configured Compute Library objects
+    auto *acl_resource
+            = ctx.get_resource_mapper()->get<acl_ip_resource_t>(this);
+    acl_ip_obj_t &acl_obj = acl_resource->get_acl_obj();
+
+    acl_obj.src_tensor.allocator()->import_memory(
```

Comment:
Do you really need to allocate and free memory each execution? Does acl have something like setting a pointer to allocate it once and then set pointer each execution and destroy tensor objects at primitive destruction?

Reply:
The memory is allocated once when `acl_ip_resource_t` is created. What `import_memory()` function actually does is `acl_ptr = onednn_ptr`, and `free()` function sets `acl_ptr = nullptr`.

>destroy tensor objects at primitive destruction
This actually happens, RDID.

Reply:
Ok, thanks for the explanation!